[Sample] Javascript sample for ai-anomaly-detector

guinao opened the pull request:
Add detect change point, detect last point anomaly, detect entire series anomaly samples for ai-anomaly-detector

joheredi commented on sdk/anomalydetector/ai-anomaly-detector/samples/javascript/sample_detect_change_point.js:
```diff
@@ -0,0 +1,64 @@
+  }
+
+  // get change point detect results
+  client.detectChangePoint(request).then((result) => {
```
Since this is within an async function, this can use await instead of `.then`

```javascript
const result = await detectChangePoint(request);
```

guinao replied:
updated.

joheredi commented on sdk/anomalydetector/ai-anomaly-detector/samples/javascript/sample_detect_change_point.js:
```diff
@@ -0,0 +1,64 @@
+
+  // get change point detect results
+  client.detectChangePoint(request).then((result) => {
+    if(result.isChangePoint.some(function(e){return e === true;})){
```
Would you mind renaming `e` and `i` to something more meaningful? It is a bit hard to follow, and since these are samples we'd prefer to be explicit.

guinao replied:
nice suggestion, updated.

joheredi commented on sdk/anomalydetector/ai-anomaly-detector/samples/javascript/sample_detect_change_point.js:
```diff
@@ -0,0 +1,64 @@
+    if(result.isChangePoint.some(function(e){return e === true;})){
+      console.log("Change points were detected from the series at index:");
+      result.isChangePoint.forEach(function(e, i){
+        if(e === true) console.log(i);
```
It'd be nice to show in a comment a sample output of this `console.log`

Also throughout the repo we prefer to have if blocks inside curly braces even when they are single line.

guinao replied:
Updated.

guinao commented:
> Great work! It looks good, as a general comment please use `await` instead of `.then`
> 
> If possible, could you please add these same samples under `samples/typescript folder`?
> 
> Also, the README.md in `sdk/anomalydetector/ai-anomaly-detector` has 2 coming soon sections one for `Key Concepts` about the anomaly detector service and another for Examples. I would really appreciate it if you could help to fill those. For the examples section, I think you can use a snippet from one of your JS samples. (See [ai-text-analytics ](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/textanalytics/ai-text-analytics/README.md)as an example)
> 
> Thanks for taking the time to create these samples 😄

Hi Jose,

We are actually add the typescript samples. However, the typescript sdk is broken and we have reported this issue to the sdk team. We will update the samples once the sdk is updated.

Also, it's nice suggestion to enrich READMD.md file. We will cooperate with document team to enrich the readme. For this PR, we want to keep the file as it will be the same with our sdk samples.

joheredi commented:
@guinao for the typescript issue the latest dev build should have the fix. Can you please try installing  @azure/ai-anomaly-detector@dev and let me know if that unlocks the typescript samples?

guinao commented:
> Thanks for addressing the comments! Can you please format the sample files with prettier? I see a couple of missing spaces that would be fixed automatically by prettier in VSCode.
> 
> Also please try out the Latest dev build to unblock TS samples. Another alternative is to switch the import from "@azure/ai-anomaly-detector" to “../../“ which should pickup the right types and intellisense

I updated the code after prettier format.

TS is unblocked. I updated typescript samples in PR #10979 . Thanks.
